inc/functions/files.php

```diff
@@ -1317,6 +1317,32 @@ function rocket_find_wpconfig_path() {
 	return false;
 }
 
+/**
+ * Get Directory files matches regex.
+ *
+ * @since 3.6.1
+ *
+ * @param string $dir   Directory to search for files inside it.
+ * @param string $regex Regular expression for files need to be searched for.
+ * @return array|RegexIterator List of files matches this regular expression.
+ */
+function rocket_get_dir_files_by_regex( $dir, $regex ) {
+	try {
+		$iterator = new IteratorIterator(
+			new FilesystemIterator( $dir )
+		);
+	} catch ( Exception $e ) {
+		return [];
+	}
+
+	try {
+		return new RegexIterator( $iterator, $regex );
+	} catch ( Exception $e ) {
+		return [];
+	}
+
+}
```

Comment:
2 things:

1. Let's make this private, i.e. to be used only internally within Rocket. This will avoid us having to deprecate it later when it becomes a method in the filesystem module.
2. Why 2 try/catch? We could simplify by combining.

```suggestion
function _rocket_get_dir_files_by_regex( $dir, $regex ) {
	try {
		$iterator = new IteratorIterator(
			new FilesystemIterator( $dir )
		);

		return new RegexIterator( $iterator, $regex );
	} catch ( Exception $e ) {
		return [];
	}
}
```